### PR TITLE
[WIP]: feat(typelint): add type linting

### PR DESCRIPTION
**Pull request by TheLarkInn**
This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case
After spending some time working on the TypeScript checking support in webpack, and how many things its been able to spot, I wanted to also start adopting it across a few projects. It would run just like eslint, and catch type errors in Editors like vs-code and in CI. We can make checking as strict or as loose as desired (I set it to the default non-superstrict). 

### Breaking Changes
Nope!

### Additional Info
There were about 30ish errors spotted on initial run locally so I expect CI to fail, but I'll fix those individual issues in separate PR's each. 

Fixes: 
- [ ] #6 - For initial support (maybe still errors with stricter checks on)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,26 +17,28 @@
   "scripts": {
     "commitlint": "commitlint",
     "commitmsg": "commitlint -e $GIT_PARAMS",
-    "lint": "eslint --cache lib test",
+    "lint": "eslint --cache lib test && npm run type-lint",
```
Comment (TheLarkInn) [inline package.json]: apparently prettierrc told my prettier to update these indentations :/ I can revert if a problem

Comment (shellscape) [reply package.json]: yeah that should be handled by the editor and .editorconfig for json files. revert that please.

**Comment by shellscape** [comment]
@TheLarkInn is this something you're wanting to complete?

**Comment by evilebottnawi** [comment]
DEPRECATED. Please use `webpack-cli`. If any features were not implemented in `webpack-cli` feel free to create issue. 

Why deprecated `webpack-command` ?
- `webpack-cli` is very stable and have more features.
- Two CLIs are misleading for developers.
- Hard to maintain two package with same purpose.
- The author stopped developing the package.
- Most of the features are already implemented in `webpack-cli`.

Thanks for using `webpack`! We apologize for the inconvenience. In the future, we will avoid such situations.

**Pull request closed without merging**